### DISCUSSION TITLE
Failed Installation of Swift with Homebrew for Linux

### DISCUSSION THREAD

**Discussion by Gerzer** [Homebrew/discussions#751] [Linux] [answered]
I tried to install Swift with Homebrew on Debian 10 via `brew install swift`, but it produced an error. I don't know why, but the `linuxbrew-core` repository isn't accepting submission of issues for anything other than bottle requests, so this discussion  page seems to be the only relevant place to report this bug.

I did find an [existing issue](https://github.com/Homebrew/linuxbrew-core/issues/20883) in the `linuxbrew-core` repository from August 2020 that reports the exact same Homebrew bug, and that user's report, including the command output, matches exactly what I encountered, so you can just refer to that issue for the details. (If there's information that you need to reproduce or to fix the bug that isn't already present in that issue, then please respond saying so, and I'll happily provide it myself.) I would've just commented on that existing issue, but it was completely ignored and then closed and later locked for being "outdated" by a bot in December 2020.

As a side note, I'd like to offer some constructive feedback to the Homebrew community: it's way too hard to submit bug reports. The issue templates are too rigid and require that you jump through too many hoops. Also, some repositories (like the `linuxbrew-core` repository) aren't accepting any bug reports at all. I love Homebrew, and the work that you put into this free, open-source project is greatly appreciated. I was lucky enough to have a professional relationship with one of the Homebrew developers in the past, which is the only reason why I was able to provide useful bug reports on problems with what was at that time the beta version of macOS, which assisted the preparation of a functioning version of Homebrew for when that version of macOS was later fully released. The obstacles to submitting bug reports the normal way can dissuade people who don't have personal connections from helping make Homebrew even better for everyone. (If you think that this warrants further discussion, then I can create a separate thread; my primary reason for posting this is to report the Swift bug.)

**Comment by SMillerDev** [comment, marked as answer]
If swift doesn't have a linux bottle it's not supported on linux. You can request a bottle, or you'll need to help fix this with a pull request. There's just too many varying systems and not enough maintainers to provide support for systems that don't work on CI.

**Reply by Gerzer** [reply]
That's what I would've thought, but the fact that running `brew install swift` on Linux actually downloads some bottle, installs various dependencies, and attempts to install Swift means that there is some formula for Swift on Linux, even though it doesn't currently work.

**Reply by SMillerDev** [reply]
There are a lot of formula that exist on linux but don't work out of the box. There are also a lot of formula that exist on ARM but don't work out of the box. Keeping these around allows users to tinker with them and hopefully fix them for everyone eventually.

**Reply by Gerzer** [reply]
Oh, OK; that's fair. In that case, I would suggest that you add a feature in which formulae on new platforms that haven't been manually validated somehow display warnings when you try to install them to avoid making it seem like a bug.

**Reply by SMillerDev** [reply]
That would be `HOMEBREW_NO_BOTTLE_SOURCE_FALLBACK` in https://docs.brew.sh/Manpage
